.github/workflows/java-sample-app-ecr-deploy.yml

```diff
@@ -68,27 +70,104 @@ jobs:
       - name: Configure AWS Credentials
         uses: aws-actions/configure-aws-credentials@v4
         with:
-          role-to-assume: ${{ secrets.E2E_IAD_TEST_ACCOUNT_ARN }}
+          role-to-assume: arn:aws:iam::${{ env.E2E_TEST_ACCOUNT_ID }}:role/${{ env.E2E_TEST_ROLE_NAME }}
           aws-region: us-east-1
 
       - name: Retrieve account
         uses: aws-actions/aws-secretsmanager-get-secrets@v1
         with:
-          secret-ids:
+          secret-ids: |
             ACCOUNT_ID, region-account/${{ matrix.aws-region }}
+            JAVA_REMOTE_SAMPLE_APP_IMAGE, e2e-test/java-remote-sample-app-image
 
       - name: Configure AWS Credentials
         uses: aws-actions/configure-aws-credentials@v4
         with:
-          role-to-assume: arn:aws:iam::${{ env.ACCOUNT_ID }}:role/${{ secrets.E2E_TEST_ROLE_ARN }}
+          role-to-assume: arn:aws:iam::${{ env.ACCOUNT_ID }}:role/${{ env.E2E_TEST_ROLE_NAME }}
           aws-region: ${{ matrix.aws-region }}
 
-      - name: Update ECR URI
-        working-directory: sample-apps/springboot-remote-service
+      - name: Build and Upload Remote Service Image v11
+        working-directory: sample-apps/java/11+/springboot-remote-service
+        run: |
+          sed -i 's#"<ECR_IMAGE_LINK>:<TAG>"#"${{ env.E2E_TEST_ACCOUNT_ID }}.dkr.ecr.us-east-1.amazonaws.com/${{ env.JAVA_REMOTE_SAMPLE_APP_IMAGE }}:v11"#g' build.gradle.kts
+          gradle jib -P javaVersion=11
+
+  upload-java-versions-main-service-image:
+    strategy:
+      fail-fast: false
+      matrix:
+        java-version: [ '8', '17', '21', '22' ]
```

Comment:
So the jobs above this one do version 11 and this one does 8, 17, 21, and 22? Why?

The way the code is structured in this PR makes it look like 8 is the odd one out, which is I assume why you have a separate folder for 8 and another one for 11+, but this matrix says otherwise

Reply:
Java 11 is distributed to all the regions for canary testing. The rest is only needed in us-east-1 for main-build tests